src/libsystemd/sd-bus/bus-control.c

```diff
@@ -786,6 +789,57 @@ _public_ int sd_bus_get_owner_creds(sd_bus *bus, uint64_t mask, sd_bus_creds **r
                 c->mask |= SD_BUS_CREDS_SUPPLEMENTARY_GIDS;
         }
 
+        if (do_sockaddr_peer) {
+                _cleanup_free_ char *t = NULL;
+
+                assert(bus->sockaddr_size_peer >= offsetof(struct sockaddr_un, sun_path) + 1);
+                assert(bus->sockaddr_peer.sa.sa_family == AF_UNIX);
+                assert(bus->sockaddr_peer.un.sun_path[0] == 0);
+
+                /* So this is an abstract namespace socket, good. Now let's find the data we are interested in */
+                r = make_cstring(bus->sockaddr_peer.un.sun_path + 1,
+                                 bus->sockaddr_size_peer - offsetof(struct sockaddr_un, sun_path) - 1,
+                                 MAKE_CSTRING_ALLOW_TRAILING_NUL,
+                                 &t);
+                if (r == -ENOMEM)
+                        return r;
+                if (r < 0)
+                        log_debug_errno(r, "Can't extract string from peer socket address, ignoring: %m");
+                else {
+                        const char *e;
+
+                        e = strstrafter(t, "/bus/");
+                        if (!e)
+                                log_debug("Didn't find /bus/ substring in peer socket address, ignoring.");
+                        else {
+                                _cleanup_free_ char *comm = NULL, *description = NULL;
+                                const char *sl;
+
+                                sl = strchr(e, '/');
+                                if (sl) {
+                                        description = strdup(sl + 1);
+                                        if (!description)
+                                                return -ENOMEM;
+
+                                        comm = strndup(e, sl - e);
+                                } else
+                                        comm = strdup(e);
```

Comment:
I think we should ignore the case, as, at least, the string has `"/bus//"` even neither comm nor description set.